src/libs/antares/concurrency/include/antares/concurrency/concurrency.h

```diff
@@ -82,6 +95,44 @@ class FutureSet
     std::vector<TaskFuture> futures_;
 };
 
+
+namespace Detail { //implementation details
+
+/*!
+ * Utility class to wrap a callable object pointer
+ * into a copyable callable object.
+ *
+ * @tparam T the underlying callable type
+ */
+template<class T>
+class SharedCallable
+{
+public:
+    explicit SharedCallable(const std::shared_ptr<T>& functionObject) :
+            functionObject_(functionObject)
+    {
+    }
+
+    void operator()()
+    {
+        (*functionObject_)();
+    }
+
+private:
+    std::shared_ptr<T> functionObject_;
+};
+
+}
+
+template <class T>
+TaskFuture AddTask(Yuni::Job::QueueService& threadPool,
+                                 const std::shared_ptr<T>& task,
+                                 Yuni::Job::Priority priority)
+{
+    Task wrappedTask = Detail::SharedCallable<T>(task);
```

Comment:
What is type `Task` ? I don't understand how `Detail::SharedCallable<T>(task)` can be of type `Task`.

Reply:
`Task` in this namespace is just an alias for std::function<void()>, so it can be assigned any function object (as long as it's copyable).